pilot/info/storagemaps.py

```diff
@@ -0,0 +1,75 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Authors:
+# - Wen Guan, wen.guan@cern.ch, 2018
+
+
+import logging
+
+from pilot.common import exception
+from pilot.info import infosys
+
+logger = logging.getLogger(__name__)
+
+
+class StorageMaps(object):
```

Comment:
Why to complicate too much the logic?  for me it looks really complicated using dedicated singletone class,  many extra wrapper functions, hundreds line of code.. why?:)
in order to map/translate `ddm.storage_id` into `ddm.name` it's just enough to simply use temporary dictionary at upper level and one line of code to look up the value in order to properly construct/patch `files` data.


the translation

Reply:
The map between ddm.name and storage_id is called very frequently.  So I don't want to go through the big ddmconf dictionary every time. I want to use it like a cache. Then I don't need to search it every time.
At first,

Reply:
In esmerge job, there can be hundreds of small input files which have only storage_id in the filespec. It's too complicated if every time to load full ddmconf data and then go through every ddm data to find whose id matches current storage_id.

Reply:
do it only once, keep the cache at higher lever -  Executor/copytool/whatever,
 Moreover the `infosys` does already cache data. 

so that  you in this case just developed singleton for dictionary-like operations, which is definitely redundant for me.

Reply:
So we still need a cache for it, just at different places. This cache will be used at different places. If I put it in Executor or copytool, it will not be convenient for me to use it in other places. A separate cache can be directly used at different places is more convenient for me. In fact, if you check the code, only few lines are real codes. others are comments or logs(in some cases, for example to log something, I split the code to more lines and added a separate return to it). If I remove all comments and logs, it can be in 15 lines. It's not complicated.

Reply:
the main issue I see is that as I already wrote before in separate MR, instead of proper constructing data from the beginning  (`files` in this case) you do complicate the logic..

anyway 15 lines vs 1 line is 15-times complication in min, plus using new redundant classes, files and codes..

less code, less errors -> clean logic and workflow.
to resolve `storage_id` by its `name`:
```python
ddm = infosys.resolve_storage_data().get(ddmendpoint)
storage_id = ddm.pk if ddm else None
```
to do back look up of ddmendpoint name by `storage_id`, it can be added to the FileSpec as info field.

Reply:
adding storage_id to filespec as a field?
It's good and it's also what I did in last pull request.  But in last request, you complained it a lot about using both storage_id and ddmendpoint in filespec.
storage_id is used only by ES. So maybe it's good not to add it to filespec.